Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino

```diff
@@ -6,70 +6,85 @@
 
-#include <OneWire.h>
-#include <DallasTemperature.h>
-#define ONE_WIRE_BUS A1 // pin for the DS18B20 data line
-OneWire oneWire(ONE_WIRE_BUS); // create a OneWire instance to communicate with the senso
-DallasTemperature sensors(&oneWire); // pass oneWire reference to Dallas Temperature sensor
-
-// Define the PWM pins for the stir bar motor
-const int stirPin1 = 5;
-const int stirPin2 = 6;
-
-// variable to store temperature
-float temperatureC;
-// KALMAN FILTER variables
-float x_k; // State estimate: current best estimate of true temp
-float p_k; // Error covariance: reps the uncertainity in the state estimate
-// Process noise and measurement noise
-float q; // Process noise covariance
-float r; // Measurement noise covariance
+#include <Wire.h>
+#include <I2Cdev.h>
+#include <MPU6050.h>
+
+MPU6050 mpu;
+
+// variables
+float initialAngle = 0.0;
+float angle = 0.0;
+float bias = 0.0;
+float P[2][2] = {{1, 0}, {0, 1}};
+float Q_angle = 0.001; // process noise covariance for the accelerometer
+float R_measure = 0.1; // measurement noise covariance
 
 void setup() {
-  Serial.begin(9600);  // start serial communication (adjust baud rate as needed)
-  sensors.begin();   // initialize the DS18B20 sensor
-
-  // Initialize Kalman filter parameters
-  x_k = 0.0; // Initial state estimate
-  p_k = 1.0; // Initial error covariance
-  q = 0.01;  // Process noise covariance 
-  r = 0.1;   // Measurement noise covariance 
-  
-  // Initialize the motor pins as outputs
-  pinMode(stirPin1, OUTPUT);
-  pinMode(stirPin2, OUTPUT);
-
-  // Set the initial speed to 80%
-  analogWrite(stirPin1, 204); // 80% of 255
-  digitalWrite(stirPin2, LOW); //for fast decay
+  Serial.begin(115200);
+
+  // initialize MPU6050
+  Wire.begin();
+  mpu.initialize();
+
+  // verify connection
+  if (!mpu.testConnection()) {
+    Serial.println("MPU6050 connection failed!");
+    while (1);
+  }
+
+  delay(1000); // let the sensor stabilize
+  initialAngle = calculateZAxisAngle();
+  Serial.print("Initial Angle: ");
+  Serial.println(initialAngle);
 }
 
 void loop() {
-  sensors.requestTemperatures();   // request temperature from all devices on the bus
-  temperatureC = sensors.getTempCByIndex(0); // get temperature in Celsius
+  // Calculate the Z-axis angle
+  float rawAngle = calculateZAxisAngle();
+
+  // Kalman filter for noise reduction
+  KalmanFilter(rawAngle);
+
+  // Print the filtered angle to the serial monitor
+  Serial.print("Filtered Angle: ");
+  Serial.println(angle);
+
+  delay(1000); // Adjust the delay based on your required polling frequency
+}
+
+float calculateZAxisAngle() {
+  int16_t ax, ay, az, gx, gy, gz;
+  mpu.getMotion6(&ax, &ay, &az, &gx, &gy, &gz);
+
+  float radians = atan2(ay, ax);
+  return radians * (180.0 / M_PI);
+}
+
+void KalmanFilter(float newAngle) {
```

Comment:
I'm not entirely sure what's going on here. I think there should be a simpler way to do this. What is the purpose of the P array? Why not just have a single value for that since we're only looking at filtering a single value? Using the same setup Rida did for the temperature sensor would probably be a good place to start just so we ensure consistency so we can keep naming schemes & formatting consistant.